aptos-move/framework/aptos-token/sources/property_map.move

```diff
@@ -160,6 +162,35 @@ module aptos_token::property_map {
         *property_val = value;
     }
 
+    public fun create_property_value_raw(
+        value: vector<u8>,
+        type: String
+    ): PropertyValue {
+        PropertyValue {
+            value,
+            type,
+        }
+    }
+
+    /// create a property value from generic type data
+    public fun create_property_value<T: copy>(data: &T): PropertyValue {
+        if (type_name<T>() == string::utf8(b"bool")) {
+            create_property_value_raw(bcs::to_bytes<T>(data), string::utf8(b"bool"))
+        } else if (type_name<T>() == string::utf8(b"u8")) {
+            create_property_value_raw(bcs::to_bytes<T>(data), string::utf8(b"u8"))
+        } else if (type_name<T>() == string::utf8(b"u64")) {
+            create_property_value_raw(bcs::to_bytes<T>(data), string::utf8(b"u64"))
+        } else if (type_name<T>() == string::utf8(b"u128")) {
+            create_property_value_raw(bcs::to_bytes<T>(data), string::utf8(b"u128"))
+        } else if (type_name<T>() == string::utf8(b"address")) {
+            create_property_value_raw(bcs::to_bytes<T>(data), string::utf8(b"address"))
+        } else if (type_name<T>() == string::utf8(b"0x1::string::String")) {
+            create_property_value_raw(bcs::to_bytes<T>(data), string::utf8(b"0x1::string::String"))
+        } else {
+            create_property_value_raw(bcs::to_bytes<T>(data), string::utf8(b"vector<u8>"))
+        }
```

Comment:
might suggest that we do something slightly different here:
```
const address_type = "address";
...

let typename = type_name<T>();

if typename == address_type || typename == bool_type {
    create_property_value(bcs::to_bytes<T>(data), typename)
} else {
    create_property_value(bcs::to_bytes<T>(data), vector<u8>)
}
```

Though it isn't clear to me why we want to not allow for arbitrary types to be stored here? Someone could just directly write whatever they want here -- or do we protect the set of types for property type?

Reply:
we only support primitive and string deserialization in PropertyMap and want to be make the serialization consistent with deserialization.